### PR TITLE
Added spacing to the 'Content - Type' header

### PR DESCRIPTION
**Pull request by tristan-fickes-bfg**
When making requests the 'Content - Type' header MUST have spaces around the hyphen otherwise the request will fail on a 400 error.  Added a new keyword search test.
**Correction: Now the 'Content-Type' shouldn't have spaces.  The spaces change how the Requests library sends the information.  We had to update how information is being passed to the Requests library which now requires the 'Content-Type' to not have spaces.

### CODE REVIEW

File: gamebench_api_client/api/requests_retriever/builder/attributes/attributes_creator.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def _get_session_headers(self):
             the appropriate headers.
 
             For most session requests, this will include the auth token and the 'accept' key/value pair.
-            Requests that don't need a session_id will also include the 'Content-Type'
+            Requests that don't need a session_id will also include the 'Content - Type'
```
Comment (josephlosee) [inline gamebench_api_client/api/requests_retriever/builder/attributes/attributes_creator.py]: Content-Type shouldn't be spaced

File: tests/test_integration/test_api/test_response/test_response_mediator.py [hunk 2]
```diff
@@ -155,12 +155,12 @@ def setUp(self):
...
                 'session_id': SESSION_ID,
                 'url': BASE_URL + VERSION + "/sessions/session_id",
-                "data": {'test_data': 'test_data'},
+                "data": '{"test_data": "test_data"}',
                 "headers": DEFAULT_SESSION_HEADERS,
                 "attributes": {
                     'params': 'test_params',
                     'headers': DEFAULT_SESSION_HEADERS,
-                    'data': {'test_data': 'test_data'}
+                    'data': '{"test_data": "test_data"}'
```
Comment (josephlosee) [inline tests/test_integration/test_api/test_response/test_response_mediator.py]: There's a lot of duplication of '{"test_data": "test_data"}', let's get that into a shared variable to make it easier to change in bulk in the future.

File: tests/test_integration/test_models/test_creator/test_model_creator.py [hunk 3]
```diff
@@ -105,3 +105,65 @@ def test_session_summary_details(self, mock_return):
         actual = model.app
 
         assert_frame_equal(actual, expected)
+
+    @requests_mock.Mocker()
+    def test_keyword_search(self, mock_return):
+        """ Keyword search should return the proper DataFrames."""
```
Comment (josephlosee) [inline tests/test_integration/test_models/test_creator/test_model_creator.py]: Please add a reference to the issue number in the docstring

File: tests/test_integration/test_models/test_creator/test_model_creator.py [hunk 4]
```diff
@@ -105,3 +105,65 @@ def test_session_summary_details(self, mock_return):
...
+        model = creator.get_model()
+        expected = pandas.DataFrame(
+            [self.keyword_search_json['response']]
+        )
+        actual = model.data
+
+        assert_frame_equal(actual, expected)
+
+    @requests_mock.Mocker()
+    def test_data_search(self, mock_return):
```
Comment (josephlosee) [inline tests/test_integration/test_models/test_creator/test_model_creator.py]: rename method to "test_apps_search"

File: gamebench_api_client/api/requests_retriever/builder/attributes/attributes_creator.py (hunk 1 above)
Comment (tristan-fickes-bfg) [reply gamebench_api_client/api/requests_retriever/builder/attributes/attributes_creator.py]: Fixed

File: tests/test_integration/test_models/test_creator/test_model_creator.py (hunk 4 above)
Comment (tristan-fickes-bfg) [reply tests/test_integration/test_models/test_creator/test_model_creator.py]: Fixed

File: tests/test_integration/test_models/test_creator/test_model_creator.py (hunk 3 above)
Comment (tristan-fickes-bfg) [reply tests/test_integration/test_models/test_creator/test_model_creator.py]: Fixed

File: tests/test_integration/test_api/test_response/test_response_mediator.py (hunk 2 above)
Comment (tristan-fickes-bfg) [reply tests/test_integration/test_api/test_response/test_response_mediator.py]: Fixed by removing some of the instances of this code duplication.  In some of the tests it wasn't necessary to include the data or params so I removed that from those tests.

**Pull request merged**